### ISSUE TITLE
coredns stuck at 'ContainerCreating' in k8s

### ISSUE THREAD

**Issue by weiyshay** [miekg/coredns#5423]
I am trying to set up k8s on CentOS 7.9 and met an issue with the latest flannel which I opened an issue at flannel which
can be resolved by rolling back flannel  from 0.18 to 0.17: 

https://github.com/flannel-io/flannel/issues/1574

After that, coredns stuck at 'ContainerCreating' for a night and I didn't see any obvious error.
Appreciate if someone can help to check what is the problem with my env?

Kubernetes version (if used): kubelet-1.23.6 kubeadm-1.23.6 kubectl-1.23.6

#kubectl get pods -A
NAMESPACE NAME READY STATUS RESTARTS AGE
kube-system coredns-6d8c4cb4d-js6rs 0/1 ContainerCreating 0 7m42s
kube-system coredns-6d8c4cb4d-tcvtt 0/1 ContainerCreating 0 7m42s
kube-system etcd-vm2-centos79 1/1 Running 0 7m53s
kube-system kube-apiserver-vm2-centos79 1/1 Running 0 7m53s
kube-system kube-controller-manager-vm2-centos79 1/1 Running 0 7m54s
kube-system kube-flannel-ds-cll2z 1/1 Running 0 7m43s
kube-system kube-proxy-42x5s 1/1 Running 0 7m43s
kube-system kube-scheduler-vm2-centos79 1/1 Running 0 7m53s

#kubectl logs -n kube-system coredns-6d8c4cb4d-js6rs
Error from server (BadRequest): container "coredns" in pod "coredns-6d8c4cb4d-js6rs" is waiting to start: ContainerCreating

# kubectl describe -n kube-system pod coredns-6d8c4cb4d-js6rs
Name:                 coredns-6d8c4cb4d-js6rs
Namespace:            kube-system
Priority:             2000000000
Priority Class Name:  system-cluster-critical
Node:                 vm2-centos79/192.168.2.12
Start Time:           Thu, 02 Jun 2022 09:58:19 -0400
Labels:               k8s-app=kube-dns
                      pod-template-hash=6d8c4cb4d
Annotations:          <none>
Status:               Pending
IP:                   
IPs:                  <none>
Controlled By:        ReplicaSet/coredns-6d8c4cb4d
Containers:
  coredns:
    Container ID:  
    Image:         registry.aliyuncs.com/google_containers/coredns:v1.8.6
    Image ID:      
    Ports:         53/UDP, 53/TCP, 9153/TCP
    Host Ports:    0/UDP, 0/TCP, 0/TCP
    Args:
      -conf
      /etc/coredns/Corefile
    State:          Waiting
      Reason:       ContainerCreating
    Ready:          False
    Restart Count:  0
    Limits:
      memory:  170Mi
    Requests:
      cpu:        100m
      memory:     70Mi
    Liveness:     http-get http://:8080/health delay=60s timeout=5s period=10s #success=1 #failure=5
    Readiness:    http-get http://:8181/ready delay=0s timeout=1s period=10s #success=1 #failure=3
    Environment:  <none>
    Mounts:
      /etc/coredns from config-volume (ro)
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-xfsc8 (ro)
Conditions:
  Type              Status
  Initialized       True 
  Ready             False 
  ContainersReady   False 
  PodScheduled      True 
Volumes:
  config-volume:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      coredns
    Optional:  false
  kube-api-access-xfsc8:
    Type:                    Projected (a volume that contains injected data from multiple sources)
    TokenExpirationSeconds:  3607
    ConfigMapName:           kube-root-ca.crt
    ConfigMapOptional:       <nil>
    DownwardAPI:             true
QoS Class:                   Burstable
Node-Selectors:              kubernetes.io/os=linux
Tolerations:                 CriticalAddonsOnly op=Exists
                             node-role.kubernetes.io/control-plane:NoSchedule
                             node-role.kubernetes.io/master:NoSchedule
                             node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                             node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Events:
  Type    Reason          Age                    From     Message
  ----    ------          ----                   ----     -------
  Normal  SandboxChanged  2m6s (x1503 over 12h)  kubelet  Pod sandbox changed, it will be killed and re-created.

**Comment by chrisohaver** [comment]
Try googling “kubelet Pod sandbox changed, it will be killed and re-created”.

**Comment by chrisohaver** [comment]
Closing this since it's a problem with other Kubernetes infrastructure, not CoreDNS.

**Comment by weiyshay** [comment]
> Try googling “kubelet Pod sandbox changed, it will be killed and re-created”.

I searched around and it seems that it's not the same issue as you mentions.
The previous issue on flannnedl commnunity was caused by an update in the 'netlink' interface change which
broke CentOS 7. I have rolled back to the older version which has been verified by others before.

Can anyone give any idea on how to debug this issue? like under what kind of situation coredns will be waiting?
